core/src/main/java/org/apache/brooklyn/feed/http/HttpFeed.java

```diff
@@ -197,6 +197,10 @@ public Builder credentials(String username, String password) {
             this.credentials = new UsernamePasswordCredentials(username, password);
             return this;
         }
+        public Builder credentials(Credentials credentials) {
```

Comment:
Agree that we should avoid it if possible. Depending on the underlying implementation being mutable and changing it **after** being passed (that's how I understand the change) doesn't sound like a good idea.